newsfragments/3248.bugfix.rst

```diff
@@ -0,0 +1 @@
+Allow pickling `trio.Cancelled` objects, as they can show up when you want to pickle something else.
```

Comment:
now that it's generalized we should update the newsfragment as well, maybe something like
```suggestion
Allow pickling `trio.Cancelled`, as they can show up when you want to pickle something else. Also enables pickling for other ``NoPublicConstructor`` objects.
```